### PR TITLE
xgboost win probability model implementation

### PR DESCRIPTION
**Pull request by lem0000ns**
**trained 4 different models:**
- logistic regression
- random forest
- xgboost (also calibrated version)
- neural network (3 layers, 2 dropout layers)

**espn wp: warriors v spurs 2/11/2026**
***win probability shown for spurs (ignore home team label)***
Screenshot 2026-02-12 at 10 02 48 PM

**xg_calibrated**
Screenshot 2026-02-12 at 10 06 21 PM

**xgboost**
Screenshot 2026-02-12 at 10 05 08 PM

**logistic regression**
Screenshot 2026-02-12 at 10 05 27 PM

**random forest**
Screenshot 2026-02-12 at 10 07 03 PM

**neural network**
Screenshot 2026-02-12 at 10 04 13 PM

**Current model**
current model used in game prediction shown on frontend is neural network. to change it, edit line 19 in backend/util.py to the corresponding model path for the desired model

### CODE REVIEW

**Review by logan-melgoza — APPROVED** [review]
LGTM
